src/index.js

```diff
@@ -36,22 +25,88 @@ btnDeleteAll.setAttribute('class', 'delete-all');
 container.append(btnDeleteAll);
 
+const addTask = (text) => {
+  const newTask = {
+    text,
+    completed: false,
+    index: todoItems.length + 1,
+  };
+  todoItems.push(newTask);
+  localStorage.setItem('todoItems', JSON.stringify(todoItems));
+};
+
 const display = () => {
   todoItems.sort((a, b) => a.index - b.index);
+  ul.innerHTML = ''; // clear the list before re-rendering it
   for (let i = 0; i < todoItems.length; i += 1) {
     const node = document.createElement('li');
-
     node.setAttribute('class', 'todo-item');
     node.setAttribute('data-key', todoItems[i].index);
 
     node.innerHTML = ` 
         <input class="checkbox" id="${todoItems[i].index}" type="checkbox"/>
-        <label for="${todoItems[i].index}" class="option"><img class="select"  src=${select2} alt=""/> <img class="delete"  src=${delete2} alt=""/></label>
+        <label for="${todoItems[i].index}" class="option">
+            <img class="select" src=${select2} alt=""/>
+            <img class="delete" src=${delete2} alt=""/>
+            <img class="edit" src=${edit} alt="edittext"/>
+        </label>
         <span class="items">${todoItems[i].text}</span>
-    
-        `;
+    `;
     ul.append(node);
+
+    // Get the selectdots, deleteoption, and editoption elements for this task
+    const selectdots = node.querySelector('.select');
+    const deleteoption = node.querySelector('.delete');
+    const editoption = node.querySelector('.edit');
+
+    // Add event listeners to show the delete/edit options when the user clicks on the selectdots
+    selectdots.addEventListener('click', () => {
+      selectdots.style.display = 'none';
+      deleteoption.style.display = 'block';
+      editoption.style.display = 'block';
+    });
+
+    // Add event listener for deleting a task
+    deleteoption.addEventListener('click', () => {
+      const itemKey = parseInt(node.getAttribute('data-key'), 10);
+      const itemIndex = todoItems.findIndex((item) => item.index === itemKey);
+      todoItems.splice(itemIndex, 1);
+
+      // Re-index the remaining tasks
+      for (let i = itemIndex; i < todoItems.length; i += 1) {
+        todoItems[i].index = i + 1;
+      }
+
+      localStorage.setItem('todoItems', JSON.stringify(todoItems));
+      display();
+    });
+
+    // Add event listener for editing a task
+    editoption.addEventListener('click', () => {
+      const itemKey = parseInt(node.getAttribute('data-key'), 10);
+      const itemIndex = todoItems.findIndex((item) => item.index === itemKey);
+      const currentTaskText = todoItems[itemIndex].text; // get the current task text
+      const newTaskText = prompt('Please enter the new task description:', currentTaskText); // pre-fill prompt with current task text
+      if (newTaskText !== null && newTaskText !== '') {
+        todoItems[itemIndex].text = newTaskText;
+        localStorage.setItem('todoItems', JSON.stringify(todoItems));
+        display();
+      }
+    });
```

Comment:
- [ ] You have very good logic for all the features, kudos for those comments! 🤓 However, please keep in mind that one of the goals of this project is to follow a given project example and try to recreate it as closely as possible!

Please avoid using prompts at all costs, they're impossible to style, stop code execution, and display differently on different browsers. According to the video example, when the user clicks on the three dots, the task description works as an input field and once you press [ENTER] the new description gets saved. Please consider implementing it this way! 🤓 🔧 


Video: https://www.youtube.com/watch?v=AcUd-_Yjjqg

Reply:
updated please see the comment and screen shot below